src/components/BookingModal/BookingModalForm/container.js

```diff
@@ -146,11 +166,22 @@ const Container = Wrapped =>
       });
     }
 
+    shouldDisableUpdateButton = formData => {
+      const {  booking,  availableDays } = this.props;
+      const originalBookingLength = getDurationBetweenDates(booking.start, booking.end);
+      const formBookingLength = getDurationBetweenDates(formData.start, formData.end);
+      const daysRemaining = availableDays + originalBookingLength;
+      const validBooking = formBookingLength <= daysRemaining;
+      return !validBooking;
```

Comment:
Good point. I'd maybe expand on the method name a bit more and call it "isValidUpdateBooking". Reason being the logic inside is specific to updating a holiday rather than creating it.

Reply:
Or possibly an even better approach would be to extend the function to work for updating and creating a holiday, depending on what arguments are passed.

Reply:
Would there be a big difference between the arguments of a created and updated holiday?

Reply:
@UnoJonnyKhattak When you're updating a holiday you need to take into consideration the available holidays + however many days were in the original booking, whereas with a new booking you're just taking into consideration the available holidays. Thinking about it, maybe extra arguments aren't even required - a check to see what the `booking` object contains to figure out if we're updating a holiday may suffice.